scripts/mirror.py

```diff
@@ -0,0 +1,61 @@
+"""
+Copy all files from the public sources in a catalog the current deployment's
+mirroring bucket. The actual file-copying is not yet implemented, so all this
+currently does is send messages to the indexer app that don't do anything.
+"""
+import argparse
+import logging
+import sys
+
+from azul import (
+    CatalogName,
+    config,
+)
+from azul.args import (
+    AzulArgumentHelpFormatter,
+)
+from azul.azulclient import (
+    AzulClient,
+)
+from azul.logging import (
+    configure_script_logging,
+)
+
+log = logging.getLogger(__name__)
+
+
+def mirror_catalog(catalog: CatalogName, wait: bool):
+    azul = AzulClient()
+    plugin = azul.repository_plugin(catalog)
+    if azul.is_queue_empty(config.mirror_queue.name):
+        fail_queue = config.mirror_queue.fail.name
+        if not azul.is_queue_empty(fail_queue):
+            log.warning('Failed messages from a previous operation are still '
+                        'present in %r. If they are not purged, this operation'
+                        'may exit with an error status, even if no new errors '
+                        'occur.', fail_queue)
+        public_sources = plugin.list_sources(authentication=None)
+        azul.remote_mirror(catalog, public_sources)
+        if wait:
+            azul.wait_for_mirroring()
+            if not azul.is_queue_empty(fail_queue):
+                log.error('Failures occurred: there are messages in %r', fail_queue)
+                sys.exit(1)
+    else:
+        log.error('A mirroring operation is already in progress. The current'
+                  'operation must finish before another can begin.')
+        sys.exit(1)
+
+
+def main(args):
```

Comment:
Should this script be allowed to run if mirroring is not enabled on the deployment?

Reply:
If mirroring is disabled, the script will fail when it tries to send messages to a nonexistent queue. But adding an additional check will make the cause of the error clearer.